pkg/util/system/socket.go

```diff
@@ -0,0 +1,55 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build !windows
+
+package system
+
+import (
+	"errors"
+	"net"
+	"os"
+	"time"
+)
+
+// CheckSocketAvailable returns if a socket at path is available
+// first boolean returns if socket path exists
+// second boolean returns if socket is reachable
+func CheckSocketAvailable(path string, timeout time.Duration) (bool, bool) {
```

Comment:
maybe `CheckSocketAccessible` instead. but no strong opinion

Reply:
`Accessible` may suggest that we only check that path is accessible (e.g. `stat` as we did before), that's why I put `Available` but also no strong opinion there.